src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRepositoryClientImplTest.java

```diff
@@ -0,0 +1,132 @@
+
+    private final FakeRemoteHttpServer mockExecutor = new FakeRemoteHttpServer();
+    @Mock
+    private BitbucketSCMRepository bitbucketSCMRepo;
+    @Mock
+    private BitbucketCICapabilities ciCapabilities;
+    private BitbucketRepositoryClientImpl repositoryClient;
+
+    @Before
+    public void setup() {
+        BitbucketRequestExecutor executor = new BitbucketRequestExecutor(BITBUCKET_BASE_URL,
+                new HttpRequestExecutorImpl(mockExecutor), OBJECT_MAPPER, BitbucketCredentials.ANONYMOUS_CREDENTIALS);
+        repositoryClient = new BitbucketRepositoryClientImpl(executor, PROJECT_KEY, REPO_SLUG);
+        when(ciCapabilities.supportsRichBuildStatus()).thenReturn(false);
+        when(bitbucketSCMRepo.getProjectKey()).thenReturn(PROJECT_KEY);
+        when(bitbucketSCMRepo.getRepositorySlug()).thenReturn(REPO_SLUG);
+    }
+
+    @Test
+    public void testPostBuildStatus() throws IOException {
+        String postURL = "http://localhost:8080/jenkins/job/Local%20BBS%20Project/15/display/redirect";
+        BitbucketBuildStatus buildStatus = new BitbucketBuildStatus.Builder("15", BuildState.INPROGRESS, postURL)
+                .setName("Local BBS Project")
+                .setDescription("#15 in progress")
+                .build();
+
+        String url = BITBUCKET_BASE_URL + "/rest/build-status/1.0/commits/" + REVISION;
+        String requestString = readFileToString("/build-status-request.json");
+        mockExecutor.mapPostRequestToResult(url, requestString, "");
+        Buffer b = new Buffer();
+
+        BitbucketBuildStatusClient client = repositoryClient.getBuildStatusClient(REVISION, ciCapabilities);
+        client.post(buildStatus);
+
+        Request clientRequest = mockExecutor.getRequest(url);
+        clientRequest.body().writeTo(b);
+        assertEquals(StringUtils.deleteWhitespace(requestString), StringUtils.deleteWhitespace(new String(b.readByteArray())));
+    }
+
+    @Test
+    public void testPostBuildStatusModernClient() throws IOException {
+        String postURL = "http://localhost:8080/jenkins/job/Local%20BBS%20Project/15/display/redirect";
+
+        InstanceKeyPairProvider keyPairProvider = mock(InstanceKeyPairProvider.class);
+        when(keyPairProvider.getPrivate()).thenReturn((RSAPrivateKey) TestUtils.createTestKeyPair().getPrivate());
+        DisplayURLProvider displayURLProvider = mock(DisplayURLProvider.class);
+        when(displayURLProvider.getRoot()).thenReturn("http://localhost:8080/jenkins");
+
+        when(ciCapabilities.supportsRichBuildStatus()).thenReturn(true);
+
+        BitbucketBuildStatus buildStatus = new BitbucketBuildStatus.Builder("15", BuildState.INPROGRESS, postURL)
+                .setName("Local BBS Project")
+                .setDescription("#15 in progress")
+                .build();
+
+        String url = String.format("%s/rest/api/1.0/projects/%s/repos/%s/commits/%s/builds", BITBUCKET_BASE_URL,
```

Comment:
As you know I'm not a fan of building URLs like this. Given this is a test I think it is probably ok.
Personally I would have opted for 
```String url = new HttpUrl.Builder()
.host(BITBUCKET_BASE_URL)
.addPathSegment(..)
...
.build()
.toString()
```

I would probably also have some helper methods that return either the `HttpUrl` or the `HttpUrl.Builder` to reduce repeating code.